### PR TITLE
Navigation

### PR DESCRIPTION
**Pull request by Ashkanthegreat**
- Make user navigation
- Make the ability to login as a user, admin, or merchant
- Make ability to log out and delete items in cart when logging out
- Make ability for merchants to see their dashboard
- Make ability for admins to see their dashboard
- Make error pages when someone tries to access not authorized pages

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        redirect_to '/profile'
+      end
+    else
+      flash[:error] = "Sorry, your credentials are bad."
+      render :new
+    end
+  end
+
+  def destroy
+    session.delete(:cart)
```
Comment (jpc20) [inline app/controllers/sessions_controller.rb:35]: Let's remember to test that this is actually emptying the cart.

File: app/models/user.rb [hunk 2]
```diff
@@ -2,4 +2,5 @@ class User < ApplicationRecord
   validates_presence_of :name, :address, :city, :state, :zip, :password
   validates :email, uniqueness: true, presence: true
   has_secure_password
+  enum role: %w(default merchant admin)
```
Comment (janegreene) [inline app/models/user.rb:5]: maps roles to numeric values

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -12,8 +12,22 @@
...
+      <% if !current_admin? %>
+        <%= link_to "Cart: #{cart.total_items}", "/cart" %>
+      <% end %>
+      <% if current_user %>
+          <%= link_to "Log out", "/logout", method: :delete %>
+      <% else %>
+        <%= link_to "Register", "/register" %>
+        <%= link_to "Login", "/login" %>
+      <% end %>
+      <% if current_merchant? %>
```
Comment (Lithnotep) [inline app/views/layouts/application.html.erb:24]: I wonder if this is the best way to identify merchants and admins in here. Possible refactor possiblity later on.

File: spec/features/users/register_new_spec.rb [hunk 4]
```diff
@@ -94,28 +94,47 @@
...
+    user = create(:user, role: 0)
+    visit "/"
+    click_on "Login"
+    fill_in :email, with: user.email
+    fill_in :password, with: user.password
+    click_on "Log In"
+
+    click_on "Log out"
+    expect(current_path).to eq(root_path)
+    expect(page).to have_content("Logged Out")
```
Comment (jpc20) [inline spec/features/users/register_new_spec.rb:138]: Lets add more expectations to confirm that we are logged out.

File: app/controllers/sessions_controller.rb [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+      redirect_to '/profile'
+      flash[:success] = "Already Logged in!"
+    end
+
+  end
+
+  def create
+    user = User.find_by(email: params[:email])
+    if user.authenticate(params[:password])
+      session[:user_id] = user.id
```
Comment (Ashkanthegreat) [inline app/controllers/sessions_controller.rb:19]: Can we refactor this?

**Pull request merged**